src/main/java/com/jlr/ttl/ds/api/controllers/VehicleController.java

```diff
@@ -12,69 +15,55 @@
 import java.util.List;
 
 @RestController
-@RequestMapping(path = "/api/load_management/vehicles")
+@AllArgsConstructor
+@RequestMapping(path = "/v1/vehicles")
 @CrossOrigin("http://localhost:4200")
 public class VehicleController {
 
-    @Autowired
     private VehicleService vehicleService;
-    @Autowired
-    private VehicleRepository vehicleRepository;
 
     @GetMapping("/all")
-    public ResponseEntity<List<Vehicle>> getAllVehicles() {
-        List<Vehicle> response = vehicleService.getAllVehiclesService();
-        return new ResponseEntity<>(response, HttpStatus.OK);
-    }
+    public ResponseEntity<DSResponse<List<VehicleResponse>>> getAllVehicles() {
+        List<VehicleResponse> dsVehicleResponse = null;
+        try{
 
-    @GetMapping("/by-vin/{vin}")
-    public ResponseEntity<?> indexVehicle(@PathVariable(value = "vin") String vin) {
-        try {
-            return ResponseEntity.ok(vehicleService.findVehicleByVinService(vin));
-        } catch (IllegalStateException e) {
-            return ResponseEntity.ok("Vehicle not existing!");
+            dsVehicleResponse = vehicleService.getAllVehicles();
+        }catch (ServiceBusinessException serviceBusinessException) {
+            return new ResponseEntity<>(DSResponse
+                    .<List<VehicleResponse>>builder()
+                    .status(DSConstants.STATUS_FAILED)
+                    .build(), HttpStatus.INTERNAL_SERVER_ERROR);
         }
+        return new ResponseEntity<>(DSResponse
+                .<List<VehicleResponse>>builder()
+                .status(DSConstants.STATUS_SUCCESS)
+                .results(dsVehicleResponse)
+                .build(), HttpStatus.OK);
     }
 
-    @GetMapping("/by-loc/{loc_code}")
-    public ResponseEntity<?> findVehiclesByLoc(@PathVariable(value = "loc_code") String loc_code) {
+    @GetMapping("/by_id/{id}")
+    public ResponseEntity<DSResponse<VehicleResponse>> getVehicleById(@PathVariable(value = "id") String id){
+
+        VehicleResponse vehicleResponse = null;
 
         try {
-            return ResponseEntity.ok(vehicleService.findVehiclesByLocService(loc_code));
-        } catch (IllegalStateException e) {
-            return ResponseEntity.ok("Illegal location!");
+            vehicleResponse = vehicleService.getVehicleByID(id);
+        } catch (IllegalStateException illegalStateException) {
```

Comment:
Do you expect to catch an IllegalStateException?

Reply:
I don't understand this part. And I didn't see line 44 45 code from my last commit. Could you please explain a bit more?

Reply:
Your service layer is not throwing an IllegalStateException so I'm not sure why we are specifically catching this type of error. Can't this be handled with a generic catch(Exception ex)? Are you aware of any specific issues

Reply:
I replaced the ServiceBusinessException thrown in the VehicleService into IllegalStateException. Is that okay?